internal/resources/metal/metal_ssh_key/resource.go

```diff
@@ -13,7 +13,7 @@ import (
 	"github.com/packethost/packngo"
 )
 
-func metalSSHKeyCommonFields() map[string]*schema.Schema {
+func MetalSSHKeyCommonFields() map[string]*schema.Schema {
```

Comment:
We should avoid repeating the package name in the function names. For example, this would be `metal_ssh_key.CommonFields()`.

Reply:
I changed the CRUD callbacks to private and renamed them. I also made the schema easier to share between metal_ssh_key and metal_project_ssh_key.